src/methods/zonal.jl

```diff
@@ -131,6 +148,13 @@ end
 function _alloc_zonal(f, x, geoms, n; kw...)
     # Find first non-missing entry and count number of missing entries
     n_missing::Int = 0
+    # TODO: wrap this in try-catch, so that we can throw an intelligent error.
+    # There are a couple of scenarios where this can fail:
+    # 1. The function cannot accept an empty iterator
```

Comment:
Ugh I guess we need an `init` keyword to handle this like `reduce`

Reply:
I would rather suggest to users:
```julia
zonal(x -> func(x; init = ...), ...)
```
since propagating a kwarg down is always weird...

Reply:
Ah right of course. We need `FixKW` in base so  this is easier